src/view/editor/parts/editor-meeting-rooms.tsx

```diff
@@ -1,126 +1,107 @@
 /*
- * SPDX-FileCopyrightText: 2022 Zextras <https://www.zextras.com>
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
-import React, { ReactElement, useCallback, useEffect, useMemo, useState } from 'react';
+import React, { ReactElement, useCallback, useMemo, useState } from 'react';
 
-import { Select } from '@zextras/carbonio-design-system';
-import { compact, find, map, xorBy } from 'lodash';
+import { DropdownItem } from '@zextras/carbonio-design-system';
+import { filter, map, uniqBy } from 'lodash';
 import { useTranslation } from 'react-i18next';
 
-import {
-	EditorAvailabilityWarningRow,
-	getIsBusyAtTimeOfTheEvent
-} from './editor-availability-warning-row';
-import { ResourceCustomComponent } from './resource-custom-component';
-import { useAttendeesAvailability } from '../../../hooks/use-attendees-availability';
+import { EditorResourceComponent, Loader, normalizeResources } from './editor-resource-component';
+import { searchResources } from '../../../soap/search-resources';
 import { useAppDispatch, useAppSelector } from '../../../store/redux/hooks';
-import {
-	selectEditorAllDay,
-	selectEditorDisabled,
-	selectEditorEnd,
-	selectEditorMeetingRoom,
-	selectEditorStart,
-	selectEditorUid
-} from '../../../store/selectors/editor';
+import { selectEditorMeetingRoom } from '../../../store/selectors/editor';
 import { editEditorMeetingRoom } from '../../../store/slices/editor-slice';
-import { useMeetingRooms } from '../../../store/zustand/hooks';
+import { useAppStatusStore } from '../../../store/zustand/store';
 import { Resource } from '../../../types/editor';
 
 export const EditorMeetingRooms = ({ editorId }: { editorId: string }): ReactElement | null => {
-	const [t] = useTranslation();
-	const disabled = useAppSelector(selectEditorDisabled(editorId));
-	const meetingRoom = useAppSelector(selectEditorMeetingRoom(editorId));
-	const [selection, setSelection] = useState<Array<Resource> | undefined>(undefined);
 	const dispatch = useAppDispatch();
-	const meetingRooms = useMeetingRooms();
-	const start = useAppSelector(selectEditorStart(editorId));
-	const end = useAppSelector(selectEditorEnd(editorId));
-	const allDay = useAppSelector(selectEditorAllDay(editorId));
-	const uid = useAppSelector(selectEditorUid(editorId));
+	const [t] = useTranslation();
+
+	const meetingRoomsValue = useAppSelector(selectEditorMeetingRoom(editorId));
+
+	const meetingRoomsChipValue = useMemo(
+		() =>
+			map(
+				meetingRoomsValue,
+				(result) =>
+					({
+						id: result.id,
+						label: result.label,
+						email: result.email,
+						avatarIcon: 'BuildingOutline',
+						avatarBackground: 'transparent',
+						avatarColor: 'gray0'
+					} as const)
```

Comment:
why is this type cast required?

Reply:
This type is casted because if you don't cast const, it will evaluate in: `{..., avatarColor: string}` instead of `{..., avatarColor: 'gray0'}`